### PR TITLE
add a way to sidestep common invalid but interpretable units in CCDData

### PR DESCRIPTION
**Pull request by eteq**
This PR addresses an annoyance I've encountered several times now.  It's nominally a dupe #7608 but it's also related to #7619.

Arguably, this PR is actually counter to some of the discussion in #7619 but I'm raising it because it's not clear #7619 got resolved, and I keep running into this particular case.  So my concrete proposal in this PR is to have a very explicit set of "known bad but unambiguous" units that `CCDData` knows how to translate.  I'm not proposing we promise that this list is complete or anything like that, but instead to think of it more like the `from_name` machinery in SkyCoord, which is very convenient even if it isn't always 100% reliable.

So basically this PR sets up machinery to do this, and populates with the specific case of #7608 with the expectation that we can add more in the future as users encounter them.  I'd say they have to justify them, which I'm doing here via #7608.

cc @mwcraig @MSeifert04

### CODE REVIEW

**Comment by bsipocz** [comment]
restarted the cancelled tests, and added the label based on the review above.

**Comment by bsipocz** [comment]
Thank you @eteq!
